packages/python/src/armonik/worker/worker.py

```diff
@@ -68,9 +73,119 @@ def Process(self, request: ProcessRequest, context) -> Union[ProcessReply, None]
             return ProcessReply(output=self.processing_function(task_handler).to_message())
         except Exception as e:
             self._logger.exception(
-                f"Failed task {''.join(traceback.format_exception(type(e) ,e, e.__traceback__))}",
+                f"Failed task {''.join(traceback.format_exception(type(e), e, e.__traceback__))}",
                 exc_info=e,
             )
 
     def HealthCheck(self, request: Empty, context) -> HealthCheckReply:
         return HealthCheckReply(status=self.health_check())
+
+
+_NOT_SET = object()
+
+
+class ArmoniKWorkerWrapper:
+    def __init__(
+        self,
+        *,
+        processor: Callable[[TaskHandler], Output],
+        logger: Optional[Logger] = None,
+        worker_endpoint: Optional[str] = None,
+        agent_endpoint: Optional[str] = None,
+        channel_options: Optional[Tuple[Tuple[str, str]]] = _NOT_SET,
+    ):
+        if logger is None:
+            ClefLogger.setup_logging(logging.INFO)
+            logger = ClefLogger.getLogger("ArmoniKWorker")
+        if worker_endpoint is None:
+            worker_scheme = (
+                "unix://"
+                if os.getenv("ComputePlane__WorkerChannel__SocketType", "unixdomainsocket")
+                == "unixdomainsocket"
+                else "http://"
+            )
+            worker_endpoint = worker_scheme + os.getenv(
+                "ComputePlane__WorkerChannel__Address", "/cache/armonik_worker.sock"
+            )
+        if agent_endpoint is None:
+            agent_scheme = (
+                "unix://"
+                if os.getenv("ComputePlane__AgentChannel__SocketType", "unixdomainsocket")
+                == "unixdomainsocket"
+                else "http://"
+            )
+            agent_endpoint = agent_scheme + os.getenv(
+                "ComputePlane__AgentChannel__Address", "/cache/armonik_agent.sock"
+            )
+        if channel_options is _NOT_SET:
+            channel_options = (("grpc.default_authority", "localhost"),)
+        self.logger = logger
+        self.worker_endpoint = worker_endpoint
+        self.agent_endpoint = agent_endpoint
+        self.channel_options = channel_options
+        self.processor = processor
+
+    def __call__(self, channel: Optional[Channel] = None):
```

Comment:
I would say this function should be `run`, and let `__call__` just be a call to `self.processor` unwrapped.